### ISSUE TITLE
https://get.sdkman.io/ Installer Script

### ISSUE THREAD

**Issue by javacraft** [sdkman/sdkman-cli#702]
We require a conversation to take place in the appropriate [SDKMAN Slack](https://slack.sdkman.io) channel before raising a new Github Issue. Please note that **issues will be closed immediately if prior discussion did not take place**. We need to do this to help manage the quality and validity of Issues raised on this project.

Please tick one:
- [ ] this is a new Feature Request, a conversation was started in the [Slack CLI Development Channel](https://sdkman.slack.com/app_redirect?channel=cli-development).
- [ ] this is a new Bug Report or Issue, a conversation was started in [Slack User Issues Channel](https://sdkman.slack.com/app_redirect?channel=user-issues).

`- [X] General question -- posted in #General but no one was home.`

Please explain the Issue / Feature Request here:

I apologize for stretching the envelope of this Issue Process. I have a simple question. I noticed that the installer script downloaded from https://get.sdkman.io/ is not part of the repo. Is this intentional or an oversight?

In short, the mere existence of a custom installation directory should not be enough to abort installation. The installation script should check if said directory is empty. If so, continue with installation; if not, abort.

For example, to install SDKMAN under `/opt/sdkman` one would either have to run as `root` or change permissions on `/opt`, either of which violates several security policies in our environment. It would be easier if a sysadmin can allocate `/opt/sdkman` set with appropriate permissions and allow user to proceed with SDKMAN installation.

Just a suggestion to make SDKMAN more enterpise-friendly.

Best regards,
Frelling

**Comment by marc0der** [comment]
@javacraft closing this issue as I'm trying to minimise the amount of issues, hence the workflow. we can have a conversation in slack when I'm available.

**Comment by 53SWYj3cV8nKsSr** [comment]
@javacraft / @marc0der, when one of you get a chance do you mind following up on this thread to tell me why the installation script is not a part of the repo?

**Comment by marc0der** [comment]
I've had a look but couldn't locate your messages in slack, sorry.

SDKMAN is not merely a script (or even a set of scripts). It is backed by a bunch of microservices that form the backend. The CLI is merely the tip of the iceberg.

SDKMAN was specifically designed to be installed into the home folder, and used by that logged in user. It is not meant to be run system-wide and will break when doing so. This is by design and won't be changing anytime soon. It is not recommended to ever run as `root`. Running and installing it under the user home is also the safest possible choice, and limits what SDKMAN can do to only the home folder of the active user. Hope this helps and makes sense!
